NETProvider/src/FirebirdSql.Data.FirebirdClient/Client/Common/XsqldaMarshaler.cs

```diff
@@ -72,13 +50,14 @@ public void CleanUpNativeData(ref IntPtr pNativeData)
 					IntPtr ptr1 = GetIntPtr(pNativeData, ComputeLength(i));
 
 					// Free	sqldata	and	sqlind pointers	if needed
-					XSQLVAR sqlvar = (XSQLVAR)Marshal.PtrToStructure(ptr1, typeof(XSQLVAR));
+					XSQLVAREmpty sqlvar = (XSQLVAREmpty)Marshal.PtrToStructure(ptr1, typeof(XSQLVAREmpty));
```

Comment:
yes. It will need to add some 'trail' struct and calculate its offset related to XSQLVar begin, and only then call Marshal.PtrToStructure.